core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/helpers/QueryModelTreePrinter.java

```diff
@@ -67,6 +69,20 @@ protected void meetNode(QueryModelNode node) {
 		}
 
 		sb.append(node.getSignature());
+
+		if (node instanceof AbstractQueryModelNode) {
+			if (((AbstractQueryModelNode) node).isVariableScopeChange()) {
```

Comment:
This is really minor, but it would be slightly more robust to cast to `VariableScopeChange` instead, as there may be future (third party) algebra extensions that don't extend AbstractQueryModelNode, but _do_ implement the interface.